src/core/mac/mac.cpp

```diff
@@ -1154,6 +1156,11 @@ ThreadError Mac::ProcessReceiveSecurity(Frame &aFrame, const Address &aSrcAddr,
             mKeyManager.SetCurrentKeySequence(keySequence);
         }
     }
+    else if (keyIdMode == Frame::kKeyIdMode2)
+    {
+        // reset frame counter
+        mKeyIdMode2FrameCounter = 0;
```

Comment:
During Interop testing, notice that if we set the transmit frame counter as `0xffffffff`, NXP device seems not response with announce message after receiving announce message. For example, 9.2.17. So here I change it to an incrementing value instead of `0xffffffff`. I will have another check with NXP device to confirm this.